clients/src/main/java/org/apache/kafka/common/security/auth/KafkaPrincipal.java

```diff
@@ -104,6 +104,10 @@ public String getPrincipalType() {
         return principalType;
     }
 
+    public void tokenAuthenticated(boolean tokenAuthenticated) {
```

Comment:
Can we remove the unused constructor now?
Also, I think I mistakenly updated the hashcode method. I can submit a minor PR.

Reply:
@omkreddy Thanks for the PR. Removed the unused constructor. I updated `hashcode` as well, but just to be sure: was it intentional not to compare `tokenAuthenticated` for `KafkaPrincipal#equals`?

Reply:
Yes, it was intentional.  During authorization checks, we compare incoming session principal  with principals stored on acl cache/store.  tokenAuthenticated field has no meaning for stored acls/principals. So we are ignoring tokenAuthenticated while comparing principals.

Another option is to move tokenAuthenticated field to Session/RequestContext objects. This may require adding new tokenAuthenticated() method to Authenticator interface.
Let me know if there are any other ideas.

Reply:
I had initially moved `tokenAuthenticated()` to `RequestContext`, but then moved it back to `KafkaPrincipal`. I was thinking that we may one day want to do something different with ACLs for token-authenticated principals. Or a custom ACL authorizer could deny some operations for token-authenticated principals, so it may be useful to keep it there.